components/ws-manager/pkg/manager/create.go

```diff
@@ -256,10 +257,19 @@ func (m *Manager) createPVCForWorkspacePod(startContext *startWorkspaceContext)
 		// Specify the storageClassName when the storage class is non-empty.
 		// This way, the Kubernetes uses the default StorageClass within the cluster.
 		// Otherwise, the Kubernetes would try to request the PVC with no class.
-		pvc.Spec.StorageClassName = &PVCConfig.StorageClass
+		PVC.Spec.StorageClassName = &PVCConfig.StorageClass
 	}
 
-	return pvc, nil
+	if startContext.VolumeSnapshot != nil {
```

Comment:
I make a test a workspace-preview, I can't start a new workspace.

I trace the log and code, looks like we need to check if `startContext.VolumeSnapshot.VolumeSnapshotName != "'` as well.

```go
if startContext.VolumeSnapshot != nil && startContext.VolumeSnapshot.VolumeSnapshotName != "" {
```

Reply:
Indeed! Pushed fix. Thank you!